BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -249,37 +293,74 @@ protected override void AutoPilot(FlightCtrlState s)
 
         void UpdateStatus()
         {
+            // Update propulsion and weapon status
             bool hasRCSFore = VesselModuleRegistry.GetModules<ModuleRCS>(vessel).Any(e => e.rcsEnabled && !e.flameout && e.useThrottle);
             hasPropulsion = hasRCSFore || VesselModuleRegistry.GetModuleEngines(vessel).Any(e => (e.EngineIgnited && e.isOperational));
             hasWeapons = weaponManager.HasWeaponsAndAmmo();
 
+            // Update intervals
             if (weaponManager.incomingMissileVessel != null && updateInterval != emergencyUpdateInterval)
                 updateInterval = emergencyUpdateInterval;
             else if (weaponManager.incomingMissileVessel == null && updateInterval == emergencyUpdateInterval)
                 updateInterval = combatUpdateInterval;
 
-            if (vessel.isActiveVessel && targetVessel && (vessel.targetObject == null || vessel.targetObject.GetVessel() != targetVessel))
+            // Set evasion status
+            EvasionStatus();
+
+            // Set target as UI target
+            if (vessel.isActiveVessel && targetVessel && !targetVessel.IsMissile() && (vessel.targetObject == null || vessel.targetObject.GetVessel() != targetVessel))
             {
-                FlightGlobals.fetch.SetVesselTarget(targetVessel);
+                FlightGlobals.fetch.SetVesselTarget(targetVessel, true);
             }
 
             return;
         }
 
+        void EvasionStatus()
```

Comment:
This function needs to be called every frame, not just during the status updates.

This is really part of a larger problem. The current implementation is fine for performing simple manoeuvres, but it locks the AI into a manoeuvre for a period without the option to easily react to changing combat situations, e.g., each separate manoeuvre has to check whether the vessel is under fire or has a missile incoming, etc., and it has no synchronisation with `BDGenericAIBase.autoPilot`, which does the WM stuff before the AI stuff.
I suggest we restructure `ShipController` and `PilotLogic` to be more along the lines of the pilot AI such that the logic gets called from `AutoPilot` instead of in separate coroutines and goes along the lines of
```csharp
protected override void AutoPilot()
{
  UpdateCommonStuff(); // manoeuvreStateChanged = false, maxAcc, upDirection, reset throttle, etc.

  CheckForNonManoeuvreStateChanges(); // Combat decisions, evasion, manoeuvreStateChanged = true and set new statusMode, etc.

  manouevreTime += Time.fixedDeltaTime;
  if (manoeuvreStateChanged || manoeuvreTime > minManoeuvreTime)
  {
    switch(statusMode)
    {
      // decide what manoeuvre state to switch to and configure whatever needs configuring
      // that wasn't already configured in CheckForNonManoeuvreStateChanges,
      // including adjusting minManoeuvreTime if needed for the different manoeuvres.
    }
    UpdateStatus();
    manouevreTime = 0;
  }
  PerformCurrentManoeuvreForAPhysicsFrame(); // Possibly also simply a switch statement depending on what's required

  AddDebugMessages();
}
```
This would allow things that need to be updated every physics frame to get updated properly without having duplicate code in each `while(...) {... wait; }` block and still let manoeuvres run for a given amount of time or get interrupted.

Also, `CalculateMaxAcceleration` should probably be called whenever the vessel part count changes (possibly also for other conditions such as running out of fuel / EC / battle damage?).